src/query/queryWithCursor.ts

```diff
@@ -39,15 +39,36 @@ export async function queryWithCursor<T extends AnyObject>(
     table.cursorSecret,
   );
 
-  const {
-    LastEvaluatedKey,
-    Items,
-    ScannedCount,
-  }: QueryOutput = await dbClient.query(params).promise();
+  let itemsData = [];
+  let LastEvaluatedKeyOutput: Key = undefined;
+  let canIterate = true;
+  let scannedCountCopy = 0;
+
+  do {
```

Comment:
you need to do this only if `filter.where` is not empty

Reply:
or i can add condition in while loop

`while (LastEvaluatedKeyOutput && canIterate && params?.Limit ) ;`

Reply:
Changed, Now it will iterate only when there is `limit` in `params`